### PR TITLE
 henrhoi/get-search-reference-api-on-build

### PR DESCRIPTION
**Pull request by henrhoi**
…uilding vespa. Now only have one version of the doc. Ready to merge, if it looks good.

### CODE REVIEW

**Review by gjoranv — CHANGES_REQUESTED** [review]

File: container-search-gui/pom.xml [hunk 1]
```diff
@@ -47,6 +47,24 @@
     </dependencies>
     <build>
         <plugins>
+            <plugin>
+                <groupId>com.googlecode.maven-download-plugin</groupId>
+                <artifactId>download-maven-plugin</artifactId>
+                <version>1.3.0</version>
+                <executions>
+                    <execution>
+                        <phase>validate</phase>
```
[the remark this answers is not in the document]
Comment (henrhoi) [reply container-search-gui/pom.xml]: Using `GUIHandler.class.getClassLoader().getResourceAsStream("gui/"+this.path);` to read from resources. Then the search-api-reference would be in another directory..

I tried with 'package' and 'prepare-package', but then the reference did not get inside the `container-search-gui-jar-with-dependencies.jar`. Had to use the 'generate-sources' phase.

File: container-search-gui/.gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 /pom.xml.build
 /target
+/src/main/resources/gui/_includes/search-api-reference.html
```
Comment (bratseth) [inline container-search-gui/.gitignore]: Could you generate to target/generated-sources/ instead? That is the standard location used inn maven projects.

**Review by bratseth — CHANGES_REQUESTED** [review]
Otherwise, great

**Comment by gjoranv** [comment]
I took the liberty of adding a commit that downloads the file in the `prepare-package` phase and into the target/classes folder. If it's downloaded to `generated-sources`, we would have to add it to the project afterwards using another plugin. I verified that this is ok according to external maver expertise.

**Review by gjoranv — APPROVED** [review]

File: container-search-gui/.gitignore (hunk 2 above)
Comment (henrhoi) [reply container-search-gui/.gitignore]: Removing this ignore, when search-api-reference.html now gets placed in /target

**Comment by henrhoi** [comment]
@bratseth Just merge if everything looks good 😃

**Review by bratseth — APPROVED** [review]

**Pull request merged**